include/Transforms/Secretize/Passes.td

```diff
@@ -22,4 +22,19 @@ def Secretize : Pass<"secretize", "ModuleOp"> {
   ];
 }
 
-#endif  // INCLUDE_TRANSFORMS_SECRETIZE_SECRETIZE_TD_
+def WrapGeneric : Pass<"wrap-generic", "ModuleOp"> {
+  let summary = "Wraps regions using secret args in secret.generic bodies";
+
+  let description = [{
+    This pass wraps regions that use secret arguments in secret.generic bodies.
+
+    Secret arguments are annotated using a `secret.secret` argument attribute.
```

Comment:
As I understand it, the pass currently only supports `func.func`, and has special treatment of that op due to how the type signature of the func comes with attributes on each type.

But this description makes it sound like the op is intended to support any region-bearing operation. Is that what you intended? Maybe `func.func` is just the initial limited support? I'm don't think block arguments can have arbitrary attributes attached to them, so I don't see a way to do it generically. But maybe we could define an interface for an op like "annotated secret" that would return a list of `OpOperand` corresponding to which arguments should be made secret? And somehow we would implement that interface for `func`... though I'm not sure how that would work if we can't change `func`'s tablegen.

If not, maybe just update the docs here to say it's just supporting `func.func`.

Reply:
I asked this question about the possibility of implementing an out-of-tree interface for an upstream op: https://discord.com/channels/636084430946959380/642426447167881246/1180722773165871124

Reply:
Updated the docs! Without being able to annotate block arguments generally, this pass won't work in other cases. I think it is possible to annotate arguments in `llvm.func` but I'm not even sure that is possible.